src/base/cub_ddl_log.c

```diff
@@ -0,0 +1,1206 @@
+	{
+	  fclose (fp);
+	  cub_ddl_log_backup (ddl_audit_handle->log_filepath);
+	  fp = NULL;
+	}
+    }
+
+write_error:
+  if (fp != NULL)
+    {
+      fclose (fp);
+      fp = NULL;
+    }
+
+  is_executed_ddl = FALSE;
+  cub_ddl_log_free (FALSE);
+  fprintf (stderr, "## %s\n", msg);
+}
+
+static int
+cub_create_log_mgs (char *msg)
+{
+  int retval = 0;
+  char result[20] = { 0 };
+  struct timeval exec_end, log_time;
+  int elapsed_sec = 0;
+  int elapsed_msec = 0;
+
+  if (ddl_audit_handle == NULL)
+    {
+      return -1;
+    }
+
+  gettimeofday (&exec_end, NULL);
+  cub_timeval_diff (&ddl_audit_handle->exec_begin_time, &exec_end, &elapsed_sec, &elapsed_msec);
+  cub_ddl_log_elapsed_time (elapsed_sec, elapsed_msec);
+
+  if (ddl_audit_handle->app_name == APP_NAME_LOADDB)
+    {
+      if (ddl_audit_handle->err_code < 0)
+	{
+	  snprintf (result, sizeof (result), "ERROR:%d", ddl_audit_handle->err_code);
+	  snprintf (ddl_audit_handle->msg, DDL_LOG_MSG, "Commited count %8d, Error line %d",
+		    ddl_audit_handle->commit_count, ddl_audit_handle->file_line_number);
+	}
+      else
+	{
+	  strcpy (result, "OK");
+	}
+
+      retval = snprintf (msg, DDL_LOG_BUFFER_SIZE, "%s %d|%s|%s|%s|%s\n",
+			 ddl_audit_handle->execute_start_time,
+			 ddl_audit_handle->pid, ddl_audit_handle->user_name, result, ddl_audit_handle->msg,
+			 ddl_audit_handle->schema_file);
+    }
+  else if (ddl_audit_handle->app_name == APP_NAME_CSQL)
+    {
+      if (ddl_audit_handle->err_code < 0)
+	{
+	  snprintf (result, sizeof (result), "ERROR:%d", ddl_audit_handle->err_code);
+	  ddl_audit_handle->elapsed_time[0] = '\0';
+	}
+      else
+	{
+	  strcpy (result, "OK");
+	}
+
+      retval = snprintf (msg, DDL_LOG_BUFFER_SIZE, "%s %d|%s|%s|%s|%s|%s\n",
+			 ddl_audit_handle->execute_start_time,
+			 ddl_audit_handle->pid,
+			 ddl_audit_handle->user_name,
+			 result, ddl_audit_handle->elapsed_time, ddl_audit_handle->msg, ddl_audit_handle->sql_text);
+    }
+  else
+    {
+      if (ddl_audit_handle->auto_commit_mode == FALSE)
+	{
+	  gettimeofday (&log_time, NULL);
+	  cub_get_time_string (ddl_audit_handle->execute_start_time, &log_time);
+	}
+
+      if (ddl_audit_handle->err_code < 0)
+	{
+	  snprintf (result, sizeof (result), "ERROR:%d", ddl_audit_handle->err_code);
+	}
+      else
+	{
+	  strcpy (result, "OK");
+	}
+
+      if (ddl_audit_handle->execute_type & DDL_LOG_RUN_EXECUTE_BATCH_FUNC)
+	{
+	  snprintf (ddl_audit_handle->elapsed_time, sizeof (ddl_audit_handle->elapsed_time), "elapsed time 0.000");
+	}
+
+      retval = snprintf (msg, DDL_LOG_BUFFER_SIZE, "%s %s|%s|%s|%s|%s|%s\n",
+			 ddl_audit_handle->execute_start_time,
+			 ddl_audit_handle->ip_addr,
+			 ddl_audit_handle->user_name,
+			 result, ddl_audit_handle->elapsed_time, ddl_audit_handle->msg, ddl_audit_handle->sql_text);
+
+    }
+
+  if (retval >= DDL_LOG_BUFFER_SIZE)
+    {
+      msg[DDL_LOG_BUFFER_SIZE - 2] = '\n';
+      msg[DDL_LOG_BUFFER_SIZE - 1] = '\0';
```

Comment:
**_snprintf_** already wrote null byte ('\0'), so this line could be removed.

Reply:
If the length of msg exceeds DDL_LOG_BUFFER_SIZE,'\n' is not written.

Reply:
Sure, you need to write to '\n' to msg[DDL_LOG_BUFFER_SIZE - 2] but you don't need write null byte to the end.
Because, already did it, so you can remove 1 line
`msg[DDL_LOG_BUFFER_SIZE - 1] = '\0';`

Reply:
Thank you, I'll remove it.